cluster-autoscaler/processors/podtemplates/podtemplate_list_processor.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package podtemplates contains PodTemplate processor used to simulate
+// DaemonSet resources from a PodTemplate.
+package podtemplates
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	apiv1 "k8s.io/api/core/v1"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/apimachinery/pkg/labels"
+	ca_context "k8s.io/autoscaler/cluster-autoscaler/context"
+	client "k8s.io/client-go/kubernetes"
+	v1lister "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+const (
+	// PodTemplateDaemonSetLabelKey use as label key on PodTemplate corresponding to an extra Daemonset.
+	PodTemplateDaemonSetLabelKey = "cluster-autoscaler.kubernetes.io/daemonset-pod"
+	// PodTemplateDaemonSetLabelValueTrue use as PodTemplateDaemonSetLabelKey label value.
+	PodTemplateDaemonSetLabelValueTrue = "true"
+)
+
+// PodTemplateListProcessor processes lists of unschedulable pods.
+type PodTemplateListProcessor interface {
+	ExtraDaemonsets(context *ca_context.AutoscalingContext) ([]*appsv1.DaemonSet, error)
+	CleanUp()
+}
+
+// activePodTemplateListProcessor returning pod list generated from PodTemplate instance.
+type activePodTemplateListProcessor struct {
+	podTemplateLister v1lister.PodTemplateLister
+	store             cache.Store
+	reflector         *cache.Reflector
+
+	ctx        context.Context
+	cancelFunc func()
+}
+
+// NewActivePodTemplateListProcessor creates an instance of PodTemplateListProcessor.
+func NewActivePodTemplateListProcessor(kubeClient client.Interface) PodTemplateListProcessor {
+	internalContext, cancelFunc := context.WithCancel(context.Background())
```

Comment:
No timeout? I see that `cancel()` is called on `CleanUp` but just double checking if TO is required.